flax/linen/normalization.py

```diff
@@ -37,6 +37,13 @@ def _canonicalize_axes(rank: int, axes: Axes) -> Iterable[int]:
     axes = (axes,)
   return tuple(set([rank + axis if axis < 0 else axis for axis in axes]))
 
+def _norm_sq(x):
+  """Computes the elementwise square norm |x|^2."""
```

Comment:
This just seems to compute `x^2`? I don't see the norm being computed anywhere, or am I missing something?

Reply:
Typically you use norm/abs value to disambiguate between x^2 = (a+bi)(a+bi) vs x^2=(a+bi)(a-bi) = a^2 + b^2. For real numbers it both amounts to normal squaring indeed.

Reply:
This function is usually called `abs2` in C++ Eigen, Julia, and has recently been added to SciPy as `abs_sq`, so maybe that would be a better name

Reply:
I agree